File: app/src/main/java/in/woloo/www/review/models/ReviewListResponse.java
package in.woloo.www.review.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class ReviewListResponse {

    @SerializedName("code")
    @Expose
    private Integer code;
    @SerializedName("status")
    @Expose
    private String status;
    @SerializedName("message")
    @Expose
    private String message;
    @SerializedName("data")
    @Expose
    private Data data;

    public Integer getCode() {
        return code;
    }

    public void setCode(Integer code) {
        this.code = code;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Data getData() {
        return data;
    }

    public void setData(Data data) {
        this.data = data;
    }

    public class Data{

        @SerializedName("total_review_count")
        @Expose
        private Integer totalReviewCount;
        @SerializedName("review_count")
        @Expose
        private Integer reviewCount;
        @SerializedName("review")
        @Expose
        private List<Review> review = null;
        @SerializedName("next")
        @Expose
        private Integer next;

        public Integer getTotalReviewCount() {
            return totalReviewCount;
        }

        public void setTotalReviewCount(Integer totalReviewCount) {
            this.totalReviewCount = totalReviewCount;
        }

        public Integer getReviewCount() {
            return reviewCount;
        }

        public void setReviewCount(Integer reviewCount) {
            this.reviewCount = reviewCount;
        }

        public List<Review> getReview() {
            return review;
        }

        public void setReview(List<Review> review) {
            this.review = review;
        }

        public Integer getNext() {
            return next;
        }

        public void setNext(Integer next) {
            this.next = next;
        }
    }

    public class Review{

        @SerializedName("id")
        @Expose
        private Integer id;
        @SerializedName("user_id")
        @Expose
        private Integer userId;
        @SerializedName("woloo_id")
        @Expose
        private Integer wolooId;
        @SerializedName("rating")
        @Expose
        private Integer rating;
        @SerializedName("remarks")
        @Expose
        private Object remarks;
        @SerializedName("status")
        @Expose
        private Integer status;
        @SerializedName("created_at")
        @Expose
        private String createdAt;
        @SerializedName("updated_at")
        @Expose
        private String updatedAt;
        @SerializedName("rating_option")
        @Expose
        private String ratingOption;
        @SerializedName("review_description")
        @Expose
        private String reviewDescription;
        @SerializedName("user_details")
        @Expose
        private UserDetails userDetails;

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public Integer getUserId() {
            return userId;
        }

        public void setUserId(Integer userId) {
            this.userId = userId;
        }

        public Integer getWolooId() {
            return wolooId;
        }

        public void setWolooId(Integer wolooId) {
            this.wolooId = wolooId;
        }

        public Integer getRating() {
            return rating;
        }

        public void setRating(Integer rating) {
            this.rating = rating;
        }

        public Object getRemarks() {
            return remarks;
        }

        public void setRemarks(Object remarks) {
            this.remarks = remarks;
        }

        public Integer getStatus() {
            return status;
        }

        public void setStatus(Integer status) {
            this.status = status;
        }

        public String getCreatedAt() {
            return createdAt;
        }

        public void setCreatedAt(String createdAt) {
            this.createdAt = createdAt;
        }

        public String getUpdatedAt() {
            return updatedAt;
        }

        public void setUpdatedAt(String updatedAt) {
            this.updatedAt = updatedAt;
        }

        public String getRatingOption() {
            return ratingOption;
        }

        public void setRatingOption(String ratingOption) {
            this.ratingOption = ratingOption;
        }

        public String getReviewDescription() {
            return reviewDescription;
        }

        public void setReviewDescription(String reviewDescription) {
            this.reviewDescription = reviewDescription;
        }

        public UserDetails getUserDetails() {
            return userDetails;
        }

        public void setUserDetails(UserDetails userDetails) {
            this.userDetails = userDetails;
        }
    }

    public class UserDetails{

        @SerializedName("id")
        @Expose
        private Integer id;
        @SerializedName("name")
        @Expose
        private String name;
        @SerializedName("avatar")
        @Expose
        private String avatar;

        public Integer getId() {
            return id;
        }

        public void setId(Integer id) {
            this.id = id;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getAvatar() {
            return avatar;
        }

        public void setAvatar(String avatar) {
            this.avatar = avatar;
        }
    }

}

File: app/src/main/java/in/woloo/www/application_kotlin/model/server_response/ReviewListResponse.kt
package `in`.woloo.www.application_kotlin.model.server_response

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class ReviewListResponse {
    @SerializedName("code")
    @Expose
    var code: Int? = null

    @SerializedName("status")
    @Expose
    var status: String? = null

    @SerializedName("message")
    @Expose
    var message: String? = null

    @SerializedName("data")
    @Expose
    var data: Data? = null

    inner class Data {
        @SerializedName("total_review_count")
        @Expose
        var totalReviewCount: Int? = null

        @SerializedName("review_count")
        @Expose
        var reviewCount: Int? = null

        @SerializedName("review")
        @Expose
        var review: List<Review>? = null

        @SerializedName("next")
        @Expose
        var next: Int? = null
    }

    inner class Review {
        @SerializedName("id")
        @Expose
        var id: Int? = null

        @SerializedName("user_id")
        @Expose
        var userId: Int? = null

        @SerializedName("woloo_id")
        @Expose
        var wolooId: Int? = null

        @SerializedName("rating")
        @Expose
        var rating: Int? = null

        @SerializedName("remarks")
        @Expose
        var remarks: Any? = null

        @SerializedName("status")
        @Expose
        var status: Int? = null

        @SerializedName("created_at")
        @Expose
        var createdAt: String? = null

        @SerializedName("updated_at")
        @Expose
        var updatedAt: String? = null

        @SerializedName("rating_option")
        @Expose
        var ratingOption: String? = null

        @SerializedName("review_description")
        @Expose
        var reviewDescription: String? = null

        @SerializedName("user_details")
        @Expose
        var userDetails: UserDetails? = null
    }

    inner class UserDetails {
        @SerializedName("id")
        @Expose
        var id: Int? = null

        @SerializedName("name")
        @Expose
        var name: String? = null

        @SerializedName("avatar")
        @Expose
        var avatar: String? = null

        @SerializedName("woloo_since")
        @Expose
        var wolooMemberSince: String? = null

        @SerializedName("base_url")
        @Expose
        var baseUrl: String? = null

    }
}
